src/main/java/io/cdap/plugin/successfactors/source/service/SuccessFactorsService.java

```diff
@@ -350,23 +356,49 @@ private InputStream filterExpandedEntityData(InputStream dataStream) throws IOEx
     for (JsonNode objectNode : arrayNode) {
       String expandOption = pluginConfig.getExpandOption();
       if (expandOption.contains(SuccessFactorsUrlContainer.PROPERTY_SEPARATOR)) {
-        expandFieldList = Arrays.asList(pluginConfig.getExpandOption()
-                                          .split(SuccessFactorsUrlContainer.PROPERTY_SEPARATOR));
+        expandFieldList = Arrays.asList(expandOption.split(SuccessFactorsUrlContainer.PROPERTY_SEPARATOR));
+      } else if (expandOption.contains("/")) {
+        expandFieldList.add(Arrays.asList(expandOption.split(SuccessFactorsUrlContainer.NAV_PROPERTY_SEPARATOR))
+                              .get(0));
```

Comment:
why are we adding only first element?

Reply:
This is for handling the expand Option input, where two or more levels of nesting exist in the entity data such as  `picklistOptions/picklistLabels`
We only need the first child as it is sufficient to bring all its grandchildren

Replaced Arrays.asList with [0] 5ba462f6e79c0757c5cd2a51632dc0219ed2a944